app_flutter/lib/logic/task_grid_filter.dart

```diff
@@ -0,0 +1,232 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:collection';
+
+import 'package:flutter/material.dart';
+
+import 'package:cocoon_service/protos.dart' show CommitStatus;
+
+import '../logic/qualified_task.dart';
+import '../widgets/filter_property_sheet.dart';
+
+/// A filter object for controlling which entries are visible in the Build dashboard grid
+/// of tasks. This filter object can filter on a number of properties of the tasks including
+/// the name of the task, information on the commit that generated the tasks, and platform
+/// stage type toggles.
+class TaskGridFilter extends FilterPropertySource {
+  TaskGridFilter();
+
+  factory TaskGridFilter.fromMap(Map<String, String> valueMap) => TaskGridFilter()..applyMap(valueMap);
+
+  /// True iff all of the properties of the filter are set to their default values.
+  bool get isDefault =>
+      _allProperties.values.toList().every((ValueFilterProperty<dynamic> element) => element.isDefault);
+
+  /// Sets all properties of this filter to their default values.
+  void reset() {
+    for (final ValueFilterProperty<dynamic> property in _allProperties.values) {
+      property.reset();
+    }
+  }
+
+  /// Modifies this filter based on the values in the map. If the map is null, no changes are
+  /// made. All keys in the [valueMap] must match one of the field names of the filter properties.
+  void applyMap(Map<String, String> valueMap) {
+    if (valueMap != null) {
+      for (final MapEntry<String, String> mapEntry in valueMap.entries) {
+        _allProperties[mapEntry.key].stringValue = mapEntry.value;
+      }
+    }
+  }
+
+  final RegExpFilterProperty _taskProperty = RegExpFilterProperty(fieldName: 'taskFilter', label: 'Task Name');
+  final RegExpFilterProperty _authorProperty = RegExpFilterProperty(fieldName: 'authorFilter', label: 'Commit Author');
+  final RegExpFilterProperty _messageProperty =
+      RegExpFilterProperty(fieldName: 'messageFilter', label: 'Commit Message');
+  final RegExpFilterProperty _hashProperty = RegExpFilterProperty(fieldName: 'hashFilter', label: 'Commit Hash');
+  final BoolFilterProperty _androidProperty = BoolFilterProperty(fieldName: 'showAndroid', label: 'Android');
+  final BoolFilterProperty _iosProperty = BoolFilterProperty(fieldName: 'showIos', label: 'iOS');
+  final BoolFilterProperty _windowsProperty = BoolFilterProperty(fieldName: 'showWindows', label: 'Windows');
+  final BoolFilterProperty _cirrusProperty = BoolFilterProperty(fieldName: 'showCirrus', label: 'Cirrus');
+  final BoolFilterProperty _luciProperty = BoolFilterProperty(fieldName: 'showLuci', label: 'Luci');
+
+  // [_allProperties] is a LinkedHashMap so we can trust its iteration order
```

Comment:
Should private variables be referenced in dart docs?

Reply:
There is no documentation there. That is a code comment.